### PR TITLE
Replace SDB with HtPP in esil_dfg.c

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/analysis/esil_dfg.c [hunk 1]
```diff
@@ -179,24 +179,28 @@ static int _rv_ins_cmp(void *incoming, void *in, void *user) {
 	return rv_incoming->from - rv_in->from;
 }
 
+static EsilDFGRegVar *newEsilDFGRegVar(const RzRegItem *ri, RzGraphNode *node) {
+	EsilDFGRegVar *rv = RZ_NEW0(EsilDFGRegVar);
+	if (!rv) {
+		return NULL;
+	}
+	rv->from = ri->offset;
```
Comment (ret2libc) [inline librz/analysis/esil_dfg.c:187]: Kudos to you for understanding how those shift/or/xor mapped to this :D Did you double check they are really the same? Honestly, I couldn't figure that out :/

**Review by ret2libc — COMMENTED** [review]
Apart from that "small detail", great job! It's much better with Ht instead of SDB!

File: librz/analysis/esil_dfg.c [hunk 2]
```diff
@@ -811,53 +794,34 @@ static bool edf_consume_1_use_old_new_push_1(RzAnalysisEsil *esil, const char *o
...
 
 	// this is not exactly necessary
 	// could use RzReg-API directly in the dfg gen,
-	// but sdb as transition table is probably faster
+	// but HT as transition table is probably faster
 	RzRegItem *ri;
 	RzListIter *ator;
 	rz_list_foreach (regs->allregs, ator, ri) {
-		const ut32 from = ri->offset;
-		const ut32 to = from + ri->size - 1;
```
Comment (pelijah) [inline librz/analysis/esil_dfg.c:856]: @ret2libc ^^^

File: librz/analysis/esil_dfg.c [hunk 3]
```diff
@@ -811,53 +794,34 @@ static bool edf_consume_1_use_old_new_push_1(RzAnalysisEsil *esil, const char *o
...
-		const ut64 v = to | (((ut64)from) << 32);
```
Comment (p371jah) [inline librz/analysis/esil_dfg.c:857]: @ret2libc  ^^^ `from` - high part, `to` - low part

**Review by ret2libc — APPROVED** [review]

**Pull request merged**